astropy/time/formats.py

```diff
@@ -1345,6 +1489,23 @@ class TimeISO(TimeString):
                 '%Y-%m-%d',
                 '{year:d}-{mon:02d}-{day:02d}'))
 
+    # Define positions and starting delimiter for year, month, day, hour,
+    # minute, seconds components of an ISO time. This is used by the fast
+    # C-parser parse_ymdhms_times()
+    #
+    #  "2000-01-12 13:14:15.678"
+    #   01234567890123456789012
+    #   yyyy-mm-dd hh:mm:ss.fff
+    # Parsed as ('yyyy', '-mm', '-dd', ' hh', ':mm', ':ss', '.fff')
+    fast_parser_pars = dict(
```

Comment:
Related to the question about a structured array above: would this be clearer if defined as one? I.e., something like
```
fast_parser_pars = np.array([
    (0, 0, 3, 0),
    (ord('-'), 4, 6, 0),
    ...],
    dtype=[('delims', np.intc), ('starts', np.intc), ('stops', np.intc), 'break_allowed', np.intc)])
```

Reply:
I think this representation is significantly less clear so I'm staying with the original. Basically transposing the original structure makes it nearly impossible to correlate the parser elements to year, month, day, hour, minute, seconds. In the current representation this is pretty clear.